src/main/java/com/ldtteam/blockout/views/View.java

```diff
@@ -83,28 +82,26 @@ public void drawSelf(final int mx, final int my)
     }
 
     @Override
-    public void scrollInput(final double wheel)
+    public boolean scrollInput(final double wheel, final double mx, final double my)
     {
-        for (final Pane child : new ArrayList<>(children))
-        {
-            if (child != null)
-            {
-                child.scrollInput(wheel);
-            }
-        }
+        final int mxChild = (int) mx - x - padding;
+        final int myChild = (int) my - y - padding;
+
+        return reverseChildIterationStream().filter(child -> child.isPointInPane(mxChild, myChild))
```

Comment:
I think all the streams you're opening here are not worth it, we never got that many elements to justify the setup overhead of a stream.

Reply:
Suggests or create a better abstraction for this problem, I'm aware of streams being a bit slower than for-each loops. Mouse moving events might be a problem using this.

Reply:
I'd just use normal loops yes.